CMakeLists.txt

```
@@ -271,9 +282,19 @@ if(CELERITAS_USE_ROOT)
   find_package(ROOT 6.24 REQUIRED)
 endif()
 
-if(CELERITAS_USE_SWIG AND NOT SWIG_FOUND)
-  # Note: 4.1 is required for C++17 parsing
-  find_package(SWIG 4.1 REQUIRED)
+if(CELERITAS_USE_SWIG)
+  if(NOT SWIG_FOUND)
+    # Note: 4.1 is required for C++17 parsing
+    find_package(SWIG 4.1 REQUIRED)
+  endif()
+
+  if(CELERITAS_REAL_TYPE STREQUAL "float")
+    celeritas_error_incompatible_option(
+      "Single-precision runtime is not compatible with SWIG"
+      CELERITAS_USE_SWIG
+      OFF
+    )
+  endif()
```

```suggestion
  if(CELERITAS_REAL_TYPE STREQUAL "float")
    celeritas_error_incompatible_option(
      "Single-precision runtime is not compatible with SWIG"
      CELERITAS_USE_SWIG
      OFF
    )
  else()
    if(NOT SWIG_FOUND)
      # Note: 4.1 is required for C++17 parsing
      find_package(SWIG 4.1 REQUIRED)
    endif()
  endif()
```
No point in complaining twice if using single precision and not finding SWIG.

---

Yeah, the reason I had it this way was uniformity with the other packages that have to be loaded first to determine compatibility. But are you manually requesting SWIG? Or is it just finding too old a version on your system?

---

I dont have SWIG at all.

---

Then it should have automatically set `CELERITAS_USE_SWIG=OFF` it it wasn't specified... do you have it on in your cmake user presets?

---

No [independently, I don't use a preset].  Swig got turned on automatically when I switch my build from VecGeom to Orange ... but this is unrelated to the change and logic in this PR.

---

That shouldn't have happened. I can't think of why it would...

---

Me neither ... I'll mention it again if/when I see the issue.